### PR TITLE
Bugfixing DB/Installer (Simplify Config Part 1)

### PR DESCRIPTION
**Pull request by nupplaphil**
- default value with HiddenString
- Fixing checkDB during installation

Prerequisite for #7371

### CODE REVIEW

File: config/defaults.config.php [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 		// pass (String)
```
Comment (MrPetovan) [inline config/defaults.config.php]: Please update the type.

File: config/defaults.config.php [hunk 2]
```diff
@@ -22,7 +22,7 @@
...
 		// Database user password. Please don't use empty passwords.
-		'password' => '',
+		'password' => new \ParagonIE\HiddenString\HiddenString(''),
```
Comment (MrPetovan) [inline config/defaults.config.php]: Additionally, does this mean all local configs have to be changed to a `HiddenString`?

**Comment by nupplaphil** [comment]
I found the problem: I checked if the "hiddenstring" is empty, but even if it is empty, we do have to store them as hiddenstring because of the typed parameters

**Comment by nupplaphil** [comment]
Oh, forgot to check the installer tests..

**Comment by nupplaphil** [comment]
I do have to temporary deactivate the auto installer tests. It'lI become much more easy to test after introducing DICE in the next weeks (I want to fix the preload cache problem first)

**Pull request merged**